torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -73,8 +73,10 @@ bool is_numpy_available() {
     PyErr_Fetch(&type, &value, &traceback);
     if (auto str = value ? PyObject_Str(value) : nullptr) {
       if (auto enc_str = PyUnicode_AsEncodedString(str, "utf-8", "strict")) {
-        if (auto byte_str = PyBytes_AS_STRING(enc_str)) {
-          message += ": " + std::string(byte_str);
+        if (enc_str) {
```

Comment:
isn't this if redundant?

Reply:
PyBytes_AS_STRING does not check nullptr , I just checked the source code.

Reply:
```
if (auto enc_str = PyUnicode_AsEncodedString(str, "utf-8", "strict"))
``` 
should check nullptr though

Reply:
Encode a Unicode object and return the result as Python bytes object. encoding and errors have the same meaning as the parameters of the same name in the Unicode [encode()](https://docs.python.org/3/library/stdtypes.html#str.encode) method. The codec to be used is looked up using the Python codec registry. Return NULL if an exception was raised by the codec.

I think "strict" will flag an internal exception used by cpython. But we all know that C has no exceptions, so that NULL has to be returned.

Reply:
```
if (auto enc-str = nullptr) {
```
evaluates to false and therefore nothing inside the if statement is executed.

Reply:
Oh, you are right..